cli/command/context/list_test.go

```diff
@@ -0,0 +1,98 @@
+package context
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/command/formatter"
+	"github.com/docker/cli/cli/context/docker"
+	"gotest.tools/assert"
+	"gotest.tools/env"
+	"gotest.tools/golden"
+)
+
+func createTestContextWithKubeAndSwarm(t *testing.T, cli command.Cli, name string, orchestrator string) {
+	revert := env.Patch(t, "KUBECONFIG", "./testdata/test-kubeconfig")
+	defer revert()
+
+	err := runCreate(cli, &createOptions{
+		name:                     name,
+		defaultStackOrchestrator: orchestrator,
+		description:              "description of " + name,
+		kubernetes:               map[string]string{keyFromCurrent: "true"},
+		docker:                   map[string]string{keyHost: "https://someswarmserver"},
+	})
+	assert.NilError(t, err)
+}
+
+func TestList(t *testing.T) {
+	cli, cleanup := makeFakeCli(t)
+	defer cleanup()
+	createTestContextWithKubeAndSwarm(t, cli, "current", "all")
+	createTestContextWithKubeAndSwarm(t, cli, "other", "all")
+	cli.SetCurrentContext("current")
+	cli.OutBuffer().Reset()
+	assert.NilError(t, runList(cli, &listOptions{
+		format: formatter.TableFormatKey,
+	}))
+	golden.Assert(t, cli.OutBuffer().String(), "list.golden")
+}
+
+func TestListUnset(t *testing.T) {
+	cli, cleanup := makeFakeCli(t)
+	defer cleanup()
+	createTestContextWithKubeAndSwarm(t, cli, "current", "unset")
+	createTestContextWithKubeAndSwarm(t, cli, "other", "all")
+	cli.SetCurrentContext("current")
+	cli.OutBuffer().Reset()
+	assert.NilError(t, runList(cli, &listOptions{
+		format: formatter.TableFormatKey,
+	}))
+	golden.Assert(t, cli.OutBuffer().String(), "list.unset.golden")
+}
+
+func TestListNoContext(t *testing.T) {
```

Comment:
I noticed the `default` context still disappears after selecting a context. Can we make the `default` context always visible in `docker context ls`?

```bash
docker context ls

NAME                DESCRIPTION                               DOCKER ENDPOINT               KUBERNETES ENDPOINT   ORCHESTRATOR
default *           Current DOCKER_HOST based configuration   unix:///var/run/docker.sock                         swarm
foobar                                                        unix:///var/run/docker.sock                         


docker context use foobar
foobar
Current context is now "foobar"

docker context ls
NAME                DESCRIPTION         DOCKER ENDPOINT               KUBERNETES ENDPOINT   ORCHESTRATOR
foobar *                                unix:///var/run/docker.sock                         


docker context use default
default
Current context is now "default"

NAME                DESCRIPTION                               DOCKER ENDPOINT               KUBERNETES ENDPOINT   ORCHESTRATOR
default *           Current DOCKER_HOST based configuration   unix:///var/run/docker.sock                         swarm
foobar                                                        unix:///var/run/docker.sock                         
```